src/universal/redux/authDuck.js

```diff
@@ -31,48 +32,59 @@ export default function reducer(state = initialState, action = {}) {
   }
 }
 
-export function setAuthToken(authToken, metaProfile = {}) {
-  const profile = metaProfile || {};
-  if (!authToken) {
-    throw new Error('setAuthToken action created with undefined authToken');
-  }
-  let obj = null;
-  try {
-    obj = jwtDecode(authToken);
-  } catch (e) {
-    throw new Error(`unable to decode jwt: ${e}`);
-  }
-  cashay.create({httpTransport: new ActionHTTPTransport(authToken)});
-  if (typeof __PRODUCTION__ !== 'undefined' && __PRODUCTION__) {
-    /*
-     * Sentry error reporting meta-information. Raven object is set via SSR.
-     * See server/Html.js for how this is initialized
-     */
-    Raven.setUserContext({ // eslint-disable-line no-undef
-      id: obj.sub,
-      email: profile.email
-    });
-  }
-  return {
-    type: SET_AUTH_TOKEN,
-    payload: {
-      obj,
-      token: authToken
-    },
-    meta: {
-      analytics: {
-        eventType: EventTypes.identify,
-        eventPayload: {
-          userId: obj.sub,
-          traits: {
-            avatar: profile.picture,
-            createdAt: profile.created_at,
-            email: profile.email,
-            name: profile.name
+export function setAuthToken(authToken, newProfile) {
+  return (dispatch, getState) => {
+    const {profile: cachedProfile} = getState();
```

Comment:
what's the difference between `profile` & `cashay.entities.User[userId]` Are we possibly duplicating state?